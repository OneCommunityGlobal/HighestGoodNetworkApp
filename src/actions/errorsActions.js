import { CLEAR_ERRORS } from '../constants/errors';

const clearErrors = () => {
  return {
    type: CLEAR_ERRORS,
  };
};

export default clearErrors;
