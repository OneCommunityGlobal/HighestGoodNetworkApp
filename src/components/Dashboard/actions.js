import { createAction } from 'redux-actions';
export const updateLeaderBoardData = createAction('UPDATE_LEADER_BOARD_DATA');
