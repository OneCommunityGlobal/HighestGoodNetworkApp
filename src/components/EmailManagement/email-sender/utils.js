import { Validators } from './validation';

export function parseRecipients(recipientText) {
  if (!recipientText || typeof recipientText !== 'string') return [];
  return recipientText
    .split(/[,;\n]/)
    .map(email => email.trim())
    .filter(email => email.length > 0 && email.includes('@'));
}

export function validateEmail(email) {
  if (!email || typeof email !== 'string') return false;
  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return emailRegex.test(email.trim());
}

export function buildRenderedEmailFromTemplate(templateData, variableValues) {
  if (!templateData) return { subject: '', content: '' };

  let content = templateData.html_content || templateData.content || '';
  let subject = templateData.subject || '';

  if (Array.isArray(templateData.variables)) {
    templateData.variables.forEach(variable => {
      if (!variable || !variable.name) return;
      // Keep {{variableName}} as is if no value provided, don't replace with [variableName]
      if (!variableValues?.[variable.name]) return;

      let value = variableValues[variable.name];
      if (variable.type === 'image') {
        const extracted = variableValues?.[`${variable.name}_extracted`];
        if (extracted) value = extracted;
        else if (value) {
          const candidate = Validators.extractImageFromSource(value);
          if (candidate) value = candidate;
        }
      }
      const regex = new RegExp(`{{${variable.name}}}`, 'g');
      content = content.replace(regex, value);
      subject = subject.replace(regex, value);
    });
  }

  return { subject, content };
}
