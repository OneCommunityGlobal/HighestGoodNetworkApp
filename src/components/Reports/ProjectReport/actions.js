
import { getProjectDetail } from '../../../actions/project';
import { fetchAllMembers } from '../../../actions/projectMembers';
import { fetchAllWBS } from '../../../actions/wbs';

export const GET_PROJECT_REPORT_BEGIN = 'GET_PROJECT_REPORT_BEGIN';
export const GET_PROJECT_REPORT_END = 'GET_PROJECT_REPORT_END';

export const getProjectReportBegin = () => ({
  type: GET_PROJECT_REPORT_BEGIN
});

export const getProjectReportEnd = () => ({
  type: GET_PROJECT_REPORT_END
});

export const getProjectReport = projectId => (dispatch) => {
  dispatch(getProjectReportBegin());

  dispatch(getProjectDetail(projectId));
  dispatch(fetchAllWBS(projectId));
  dispatch(fetchAllMembers(projectId));

  dispatch(getProjectReportEnd());
}
