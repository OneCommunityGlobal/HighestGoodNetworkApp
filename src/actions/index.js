import httpService from '../services/httpService';
import axios from 'axios';
import { ApiEndpoint } from '../utils/URL';
import { ENDPOINTS } from '../utils/URL';


const APIEndpoint = ApiEndpoint;

export function clearUserProfile() {
  return { type: 'CLEAR_USER_PROFILE' };
};

export function getUserTeamMembers(userId) {
  const request = httpService.get(`${APIEndpoint}/userprofile/teammembers/${userId}`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_USER_TEAM_MEMBERS',
        payload: data,
      });
    });
  };
};

export function getUserProjectMembers(projectId) {
  const request = httpService.get(`${APIEndpoint}/userprofile/project/${projectId}`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_USER_PROJECT_MEMBERS',
        payload: data,
      });
    });
  };
};

export function getDashboardData(userId) {
  const request = httpService.get(`${APIEndpoint}/dashboard/${userId}`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_DASHBOARD_DATA',
        payload: data,
      });
    });
  };
};

export function getWeeklyDashboardData(userId, fromDate, toDate) {
  const request = httpService.get(
    `${APIEndpoint}/dashboard/weeklydata/${userId}/${fromDate}/${toDate}`,
  );

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_WEEKLY_DASHBOARD_DATA',
        payload: data,
      });
    });
  };
};

export function getMonthlyDashboardData(userId, fromDate, toDate) {
  const request = httpService.get(
    `${APIEndpoint}/dashboard/monthlydata/${userId}/${fromDate}/${toDate}`,
  );

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_MONTHLY_DASHBOARD_DATA',
        payload: data,
      });
    });
  };
};

export function getLeaderboardData(userId) {
  const request = httpService.get(`${APIEndpoint}/dashboard/leaderboard/${userId}`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_LEADERBOARD_DATA',
        payload: data,
      });
    });
  };
};

export function getActionItems(userId) {
  const request = httpService.get(`${APIEndpoint}/actionItem/user/${userId}`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_ACTION_ITEMS',
        payload: data,
      });
    });
  };
};

export function getNotifications(userId) {
  const request = httpService.get(`${APIEndpoint}/notification/user/${userId}`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_NOTIFICATIONS',
        payload: data,
      });
    });
  };
};

export function getAllProjects() {
  const request = httpService.get(`${APIEndpoint}/projects`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_ALL_PROJECTS',
        payload: data,
      });
    });
  };
};

export function getProjectById(projectId) {
  const request = httpService.get(`${APIEndpoint}/project/${projectId}`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_PROJECT_BY_ID',
        payload: data,
      });
    });
  };
};

export function getProjectsByUser(userId) {
  const request = httpService.get(`${APIEndpoint}/projects/user/${userId}`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_PROJECTS_BY_USER',
        payload: data,
      });
    });
  };
};

export function getProjectMembership(projectId) {
  const request = httpService.get(`${APIEndpoint}/project/${projectId}/users`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_PROJECT_MEMBERSHIP',
        payload: data,
      });
    });
  };
};

export function getAllTeams() {
  const request = httpService.get(`${APIEndpoint}/team`);
  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_ALL_TEAMS',
        payload: data,
      });
    });
  };
};

export function getTeamById(teamId) {
  const request = httpService.get(`${APIEndpoint}/team/${teamId}`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_TEAM_BY_ID',
        payload: data,
      });
    });
  };
};

export function getTeamMembership(teamId) {
  const request = httpService.get(`${APIEndpoint}/team/${teamId}/users`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_TEAM_MEMBERSHIP',
        payload: data,
      });
    });
  };
};

export function getAllTimeEntries() {
  const request = httpService.get(`${APIEndpoint}/TimeEntry`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_ALL_TIME_ENTRIES',
        payload: data,
      });
    });
  };
};

export function getTimeEntryForSpecifiedPeriod(userId, fromDate, toDate) {
  const request = httpService.get(`${APIEndpoint}/TimeEntry/user/${userId}/${fromDate}/${toDate}`);

  return dispatch => {
    request.then(({ data }) => {
      dispatch({
        type: 'GET_TIME_ENTRY_FOR_SPECIFIED_PERIOD',
        payload: data,
      });
    });
  };
};

export function postTimeEntry(timeEntryObj) {
  const request = httpService.post(`${APIEndpoint}/TimeEntry`, timeEntryObj);
  return dispatch => {
    request.then(
      response => dispatch({ type: 'REQUEST_SUCCEEDED', payload: response }),
      error => dispatch({ type: 'REQUEST_FAILED', error }),
    );
  };
};

export function getTimeEntryByProjectSpecifiedPeriod(projectId, fromDate, toDate) {
  const request = httpService.get(`${APIEndpoint}/TimeEntry/projects/${projectId}/${fromDate}/${toDate}`);

  return dispatch => {
    return new Promise((resolve, reject) => {
      request.then(({ data }) => {
        dispatch({
          type: 'GET_TIME_ENTRY_By_Project_FOR_SPECIFIED_PERIOD',
          payload: data,
        });
        resolve(data);
      }).catch(error => {
        reject(error);
      });
    });
  };
};

export function getTimeEntryForOverDate(users, fromDate, toDate) {

  const url = ENDPOINTS.TIME_ENTRIES_USER_LIST;

  return axios.post(url, { users, fromDate, toDate })
  .then(response => response.data)
  .catch(error => {
    throw error;
  });
}
