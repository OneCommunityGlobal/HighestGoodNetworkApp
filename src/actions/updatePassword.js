import axios from 'axios';
import { GET_ERRORS } from '../constants/errors';
import { ENDPOINTS } from '../utils/URL';

export const updatePassword = (userId, newpasswordData) => {
  const url = ENDPOINTS.UPDATE_PASSWORD(userId);
  return async (dispatch) => {
    try {
      const res = await axios.patch(url, newpasswordData);
      return res.status;
    } catch (e) {
      dispatch({
        type: GET_ERRORS,
        payload: e.response.data,
      });
      return e.response.status;
    }
  };
};

export const forcePasswordUpdate = (data) => {
  const url = ENDPOINTS.FORCE_PASSWORD;
  return async (dispatch) => {
    try {
      const res = await axios.patch(url, data);
      return res.status;
    } catch (e) {
      dispatch({
        type: GET_ERRORS,
        payload: e.response.data,
      });
      return e.response.status;
    }
  };
};
