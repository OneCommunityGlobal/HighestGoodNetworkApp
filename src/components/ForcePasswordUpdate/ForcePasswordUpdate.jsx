import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Joi from 'joi';
import { toast } from 'react-toastify';
import { forcePasswordUpdate } from '../../actions/updatePassword';
import { clearErrors } from '../../actions/errorsActions';
import Form from '../common/Form';

export class ForcePasswordUpdate extends Form {
  state = {
    data: { newpassword: '', confirmnewpassword: '' },
    errors: {},
  };

  componentDidMount() {
    // document.title = "Force Update Password";
  }

  componentDidUpdate(prevProps) {
    if (prevProps.errors.error !== this.props.errors.error) {
      this.setState({ errors: this.props.errors });
    }
  }

  componentWillUnmount() {
    this.props.clearErrors();
  }

  schema = {
    newpassword: Joi.string()
      .regex(/(?=^.{8,}$)((?=.*\d)|(?=.*\W+))(?![.\n])(?=.*[A-Z])(?=.*[a-z]).*$/)
      .required()
      .label('New Password')
      .options({
        language: {
          string: {
            regex: {
              base:
                'should be at least 8 characters long and must include at least one uppercase letter, one lowercase letter, and one number or special character',
            },
          },
        },
      }),

    confirmnewpassword: Joi.any()
      .valid(Joi.ref('newpassword'))
      .options({ language: { any: { allowOnly: 'must match new password' } } })
      .label('Confirm Password'),
  };

  doSubmit = async () => {
    const { newpassword } = {
      ...this.state.data,
    };

    const { userId } = this.props.match.params;
    const data = { userId, newpassword };
    const status = await this.props.forcePasswordUpdate(data);
    if (status === 200) {
      toast.success(
        'You will now be directed to the login page where you can login with your new password.',
        {
          onClose: () => this.props.history.replace('/login'),
        },
      );
    } else {
      toast.error('Something went wrong. Please contact your administrator.');
    }
  };

  render() {
    const { darkMode } = this.props;

    return (
      <div
        className={`pt-5 h-100 container-fluid d-flex flex-column align-items-center ${
          darkMode ? 'bg-oxford-blue' : ''
        }`}
      >
        <h2>Change Password</h2>

        <form className="col-md-4 xs-12" onSubmit={e => this.handleSubmit(e)}>
          {this.renderInput({
            name: 'newpassword',
            label: 'New Password:',
            type: 'password',
            darkMode,
          })}
          {this.renderInput({
            name: 'confirmnewpassword',
            label: 'Confirm Password:',
            type: 'password',
            'data-refers': 'newpassword',
            darkMode,
          })}
          {this.renderButton({ label: 'Submit', darkMode })}
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  errors: state.errors,
  darkMode: state.theme.darkMode,
});

export default withRouter(
  connect(mapStateToProps, {
    forcePasswordUpdate,
    clearErrors,
  })(ForcePasswordUpdate),
);
