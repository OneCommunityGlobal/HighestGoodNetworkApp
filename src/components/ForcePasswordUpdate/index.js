import ForcePasswordUpdate from './ForcePasswordUpdate';

export default ForcePasswordUpdate;
