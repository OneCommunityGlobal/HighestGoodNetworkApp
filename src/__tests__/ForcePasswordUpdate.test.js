import { renderWithProvider, renderWithRouterMatch } from './utils.js';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import mockState from './mockAdminState.js';
import { createMemoryHistory } from 'history';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { ENDPOINTS } from '../utils/URL';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import routes from './../routes';
import { ForcePasswordUpdate } from '../components/ForcePasswordUpdate/ForcePasswordUpdate';
import { forcePasswordUpdate as fPU } from './../actions/updatePassword';
import { clearErrors } from './../actions/errorsActions';
import { shallow } from 'enzyme';

describe('Force Password Update page structure', () => {
  let mountedFPUpdate, props;
  beforeEach(() => {
    props = {
      auth: { isAuthenticated: true },
      errors: {},
      clearErrors: clearErrors,
      forcePasswordUpdate: ForcePasswordUpdate,
    };
    mountedFPUpdate = shallow(<ForcePasswordUpdate {...props} />);
  });

  it('should be rendered with two input fields', () => {
    const inputs = mountedFPUpdate.find('Input');
    expect(inputs.length).toBe(2);
  });

  it('should be rendered with one button', () => {
    const button = mountedFPUpdate.find('button');
    expect(button.length).toBe(1);
  });

  it('should be rendered with one h2 labeled Change Password', () => {
    const h2 = mountedFPUpdate.find('h2');
    expect(h2.length).toEqual(1);
    expect(h2.first().text()).toContain('Change Password');
  });
});

describe('When user tries to input data', () => {
  let mountedFPUpdate, props, fPU;

  beforeEach(() => {
    fPU = jest.fn();
    props = {
      match: { params: { userId: '5edf141c78f1380017b829a6' } },
      auth: { isAuthenticated: true },
      errors: {},
      clearErrors: clearErrors,
      forcePasswordUpdate: fPU,
    };
    mountedFPUpdate = shallow(<ForcePasswordUpdate {...props} />);
  });

  it('should call handleInput when input is changed', () => {
    const spy = jest.spyOn(mountedFPUpdate.instance(), 'handleInput');
    mountedFPUpdate
      .find("[name='newpassword']")
      .simulate('change', { currentTarget: { name: 'newpassword', value: 'abc' } });
    expect(spy).toHaveBeenCalled();
  });

  it('should correctly update the new password value in the state', () => {
    const expected = 'newpassword';
    const Input = { name: 'newpassword', value: expected };
    const mockEvent = { currentTarget: Input };
    mountedFPUpdate.instance().handleInput(mockEvent);

    expect(mountedFPUpdate.instance().state.data.newpassword).toEqual(expected);
  });

  it('should correctly update the confirm password value in the state', () => {
    const expected = 'newpassword';
    const Input = { name: 'confirmnewpassword', value: expected };
    const mockEvent = { currentTarget: Input };
    mountedFPUpdate.instance().handleInput(mockEvent);

    expect(mountedFPUpdate.instance().state.data.confirmnewpassword).toEqual(expected);
  });

  it('should have disabled submit button if form is invalid', () => {
    const button = mountedFPUpdate.find('button');
    expect(button.props()).toHaveProperty('disabled');
  });

  it('onSubmit forcePasswordUpdate method is called with credentials', async () => {
    await mountedFPUpdate.instance().doSubmit();
    expect(fPU).toHaveBeenCalledWith({ userId: '5edf141c78f1380017b829a6', newpassword: '' });
  });
});

const url = ENDPOINTS.FORCE_PASSWORD;
const userProjectsUrl = ENDPOINTS.USER_PROJECTS(mockState.auth.user.userid);
let passwordUpdated = false;
//When user is sent to forced Password Update they are not
//authenticated yet and will be sent to login afterwards
mockState.auth.isAuthenticated = false;

function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

const server = setupServer(
  //request for a forced password update.
  rest.patch(url, (req, res, ctx) => {
    passwordUpdated = true;
    if (req.body.newpassword === 'newPassword8') {
      return res(ctx.status(200));
    }
  }),
  //prevents errors when loading header
  rest.get('http*/api/userprofile/*', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json({}));
  }),
  rest.get('http://*/hash.txt', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json({}));
  }),
  //Leaderboard Data in case user gets to dashboard.
  rest.get('http://localhost:4500/api/dashboard/*', (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json([
        {
          personId: '5edf141c78f1380017b829a6',
          name: 'Dev Admin',
          weeklycommittedHours: 10,
          totaltime_hrs: 6,
          totaltangibletime_hrs: 6,
          totalintangibletime_hrs: 0,
          percentagespentintangible: 100,
          didMeetWeeklyCommitment: false,
          weeklycommitted: 10,
          tangibletime: 6,
          intangibletime: 0,
          tangibletimewidth: 100,
          intangibletimewidth: 0,
          tangiblebarcolor: 'orange',
          totaltime: 6,
        },
      ]),
    );
  }),
  rest.get(userProjectsUrl, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json([
        {
          isActive: true,
          _id: '5ad91ec3590b19002acfcd26',
          projectName: 'HG Fake Project',
        },
      ]),
    );
  }),
  //Any other requests error out
  rest.get('*', (req, res, ctx) => {
    console.error(
      `Please add request handler for ${req.url.toString()} in your MSW server requests.`,
    );
    return res(ctx.status(500), ctx.json({ error: 'You must add request handler.' }));
  }),
);

beforeAll(() => server.listen());
afterAll(() => server.close());
afterEach(() => server.resetHandlers());

describe('Force Password Update behaviour', () => {
  let fPUMountedPage, rt, hist;
  beforeEach(() => {
    rt = '/forcePasswordUpdate/5edf141c78f1380017b829a6';
    hist = createMemoryHistory({ initialEntries: [rt] });
    fPUMountedPage = renderWithRouterMatch(routes, {
      initialState: mockState,
      route: rt,
      history: hist,
    });
  });

  it('should pop up an error if password doesnt meet requirements', async () => {
    let reqError =
      '"New Password" should be at least 8 characters long and must include at least one uppercase letter, one lowercase letter, and one number or special character';
    //No number or special char
    fireEvent.change(screen.getByLabelText('New Password:'), {
      target: { value: 'newPassword' },
    });

    await waitFor(() => {
      expect(passwordUpdated).toBeFalsy();
      expect(screen.getByText(reqError)).toBeTruthy();
    });

    //No capatalized char
    fireEvent.change(screen.getByLabelText('New Password:'), {
      target: { value: 'newpassword8' },
    });

    await waitFor(() => {
      expect(passwordUpdated).toBeFalsy();
      expect(screen.getByText(reqError)).toBeTruthy();
    });

    //Not long enough
    fireEvent.change(screen.getByLabelText('New Password:'), {
      target: { value: 'word8' },
    });

    await waitFor(() => {
      expect(passwordUpdated).toBeFalsy();
      expect(screen.getByText(reqError)).toBeTruthy();
    });
  });

  it('should pop up an error if passwords dont match', async () => {
    fireEvent.change(screen.getByLabelText('New Password:'), {
      target: { value: 'newPassword8' },
    });

    fireEvent.change(screen.getByLabelText('Confirm Password:'), {
      target: { value: 'Password8' },
    });

    fireEvent.click(screen.getByText('Submit'));

    await waitFor(() => {
      expect(passwordUpdated).toBeFalsy();
      expect(screen.getByText('"Confirm Password" must match new password')).toBeTruthy();
    });
  });
  it('should update password after submit is clicked', async () => {
    //const pushSpy = jest.spyOn(history, 'replace');
    const history = { replace: jest.fn() };
    fireEvent.change(screen.getByLabelText('New Password:'), {
      target: { value: 'newPassword8' },
    });

    fireEvent.change(screen.getByLabelText('Confirm Password:'), {
      target: { value: 'newPassword8' },
    });

    fireEvent.click(screen.getByText('Submit'));

    await waitFor(() => {
      expect(passwordUpdated).toBeTruthy();
      fireEvent.click(
        screen.getByText(
          'You will now be directed to the login page where you can login with your new password.',
        ),
      );
    });
    /*
    await waitFor(()=> {
      expect(screen.getByLabelText('Email:')).toBeTruthy();
    });
    */
  });
});

describe('Force Password Update page structure', () => {
  it('should match the snapshot', () => {
    let props = {
      match: { params: { userId: '5edf141c78f1380017b829a6' } },
      auth: { isAuthenticated: true },
      errors: {},
      clearErrors: clearErrors,
      forcePasswordUpdate: fPU,
    };
    const { asFragment } = render(<ForcePasswordUpdate {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
