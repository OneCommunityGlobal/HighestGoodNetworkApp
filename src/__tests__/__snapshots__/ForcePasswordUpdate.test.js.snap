// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Force Password Update page structure should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-5"
  >
    <h2>
      Change Password
    </h2>
    <form
      class="col-md-6 xs-12"
    >
      <div
        class="form-group "
      >
        <label
          for="newpassword"
        >
          New Password:
        </label>
        <input
          class="form-control"
          id="newpassword"
          name="newpassword"
          type="password"
          value=""
        />
      </div>
      <div
        class="form-group "
      >
        <label
          for="confirmnewpassword"
        >
          Confirm Password:
        </label>
        <input
          class="form-control"
          data-refers="newpassword"
          id="confirmnewpassword"
          name="confirmnewpassword"
          type="password"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        disabled=""
        style="box-shadow: 2px 2px 4px 1px lightgray;"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
